src/crypto/index.js

```diff
@@ -93,6 +94,11 @@ function Crypto(baseApis, eventEmitter, sessionStore, userId, deviceId,
 
     this._globalBlacklistUnverifiedDevices = false;
 
+    // list of IncomingRoomKeyRequests/IncomingRoomKeyRequestCancellations
+    // we received in the current sync.
+    this._receivedRoomKeyRequests = [];
+    this._receivedRoomKeyRequestCancellations = [];
```

Comment:
> we will process the key request, and then process the cancellation.

Doesn't that race? We should at least mention the ordering concerns and decisions / things we've punted on for now somewhere.